### DISCUSSION TITLE
How to create CubeNode with rounded edges?

### DISCUSSION THREAD

**Discussion by 1729patrick** [sceneview/sceneview#577] [Q&A]
Could not find a way to do it, Is it not available?

**Comment by ThomasGorisse** [comment]
v4.0.0-rc.1 provides `CubeNode` out of the box, but rounded edges require custom geometry. Two approaches:

1. **Load a GLB model** — Create a rounded cube in Blender/3D tool and export as GLB. This is the easiest and most performant approach.

2. **Custom MeshNode** — Generate vertices with beveled edges using the geometry utilities in `sceneview-core`:
```kotlin
MeshNode(
    materialInstance = material,
    vertices = roundedCubeVertices, // your generated geometry
    indices = roundedCubeIndices
)
```

For most use cases, a pre-made GLB is recommended since rounded corners require significantly more vertices than a sharp cube.
